extension/settingsproviderextension/subscribers/subscriber_hub.go

```diff
@@ -0,0 +1,122 @@
+package subscribers
+
+import (
+	stsSettingsModel "github.com/stackvista/sts-opentelemetry-collector/connector/tracetotopoconnector/generated/settings"
+	stsSettingsEvents "github.com/stackvista/sts-opentelemetry-collector/extension/settingsproviderextension/events"
+	"go.uber.org/zap"
+	"sync"
+)
+
+const (
+	defaultBufferSize = 2
+)
+
+type subscriber struct {
+	settingTypes map[stsSettingsModel.SettingType]struct{}
+	channel      chan stsSettingsEvents.UpdateSettingsEvent
+}
+
+type SubscriberHub struct {
+	logger *zap.Logger
+
+	// Mutex for concurrent access to subscribers
+	subscribersLock sync.RWMutex
+	// The list of subscribers to which signals that settings have been updated will be sent
+	subscribers []subscriber
+}
+
+func NewSubscriberHub(logger *zap.Logger) *SubscriberHub {
+	return &SubscriberHub{
+		logger:      logger,
+		subscribers: make([]subscriber, 0),
+	}
+}
+
+func (h *SubscriberHub) Register(types ...stsSettingsModel.SettingType) <-chan stsSettingsEvents.UpdateSettingsEvent {
+	return h.RegisterWithBuffer(defaultBufferSize, types...)
+}
+
+func (h *SubscriberHub) RegisterWithBuffer(bufferSize int, types ...stsSettingsModel.SettingType) <-chan stsSettingsEvents.UpdateSettingsEvent {
+	h.subscribersLock.Lock()
+	defer h.subscribersLock.Unlock()
+
+	typeSet := make(map[stsSettingsModel.SettingType]struct{}, len(types))
+	for _, t := range types {
+		typeSet[t] = struct{}{}
+	}
+
+	// buffered so sender won’t block
```

Comment:
It is anyway a non-blocking send. Thinking about the "protocol" between a subscriber and the hub it seems that we could do without a buffer and use the notification channel simply like that: a notification:
* If there is a message in the channel there have been updates for your registered types. You should fetch them to get the latest state.
* If there is no message there are no updates

It is actually a feature that there is no buffer: there will not be 3 more messages buffered repeating the information that there is an update. You, as a subscriber, already fetched the latest state, so already covered those updates too.